### PR TITLE
PE-75 | Implemented deleteUserService for PUT operation

### PR DESCRIPTION
**Pull request by txt020**
**What I did:**

- Created a service in the frontend calling the controller created in the backend by @txt020.
- Users will be able to delete their account from Firebase and "fill delete" into their account from MongoDB

**How to Test:**

1. Make sure you're logged into Firebase and Mongodb.
2. Pull from my branch and run `flutter pub get` 
3. In VSC fetch and pull from **remote main** to acquire the newest changes for Springboot.
4. Run Springboot with `mvn spring-boot:run`
5. On your device make a new local or google account and check to make sure it is in Firebase and MongoDb.
6. Login in and navigate through the tabs to delete your account. Follow Instructions until you're back at login screen.
7. Make sure your account was deleted from Firebase and MongoDb.
8. Repeat steps 6 and 7 for the google account.

### CODE REVIEW

File: lib/screens/settings/account/account_settings_delete.dart [hunk 1]
```diff
@@ -87,9 +162,7 @@ class AccountSettingsDelete extends StatelessWidget {
                 ),
               ),
               FlatButton(
-                onPressed: () {
-                  deleteUserAccount(context);
-                },
+                onPressed: () => deleteUserAccount(context),
```
Comment (ayeejane) [inline lib/screens/settings/account/account_settings_delete.dart:165]: Clicking on the **Delete** button should pop up another dialog box that lets the user know that their account has been deleted for ex: "Your account has been successfully deleted." I believe I implemented this on my old task before PE-41. It seems like you changed it up a bit that its not showing the dialog box anymore.

**Review by ayeejane — CHANGES_REQUESTED** [review]

**Comment by ayeejane** [comment]
Also, the local or google email I'm signing up with is not getting deleted from MongoDB, only Firebase.

File: lib/screens/settings/account/account_settings_delete.dart (hunk 1 above)
Comment (txt020) [reply lib/screens/settings/account/account_settings_delete.dart:165]: @ayeejane In your PE-41, there is no implementation of notifying the user their account was successfully deleted. In addition, users being redirected to login screen implies their account has been deleted. I would not recommend another dialog box popup, based off the current workflow. The user is aware that their account is going to be deleted based off the instructions.

**Comment by dominikyz** [comment]
Is there particular reason to store  bunch of deleted accounts in Mongo? 
image
As well, why not use DELETE instead of put?

When deleting google account i don't get enter password window as we get when account is manually created

File: lib/services/firebase_exception_handler.dart [hunk 2]
```diff
@@ -16,9 +16,9 @@ class FirebaseExceptionHandler {
       case 'We have blocked all requests from this device due to unusual activity. Try again later.':
         return 'Too many failed login attempts, please try again later.';
         break;
-      case 'Given String is empty or null':
+      /*case 'Given String is empty or null':
         return 'Email required.';
-        break;
+        break;*/
```
Comment (dominikyz) [inline lib/services/firebase_exception_handler.dart:21]: If we dont use it, lets clean this comment

**Review by dominikyz — CHANGES_REQUESTED** [review]
Please check comments . 
If any of my concerns in comments are just planned operations, then i'll approve pull. 
Otherwise everything works. Nice job!

**Review by ayeejane — APPROVED** [review]

File: lib/services/firebase_exception_handler.dart (hunk 2 above)
Comment (txt020) [reply lib/services/firebase_exception_handler.dart:21]: @dominikyz  This is a planned operation and will be used in later updates for the app.

**Comment by txt020** [comment]
@dominikyz  My task we to implement a service that will overwrite each user with "deleted". I believe the reason why we are doing this is to save the users ID and help identify which users needs to be deleted. 

@tonynguyen98 Feel free to chime in if I am missing anything. ^

@dominikyz The reason why you don't see a password requirement for you google account. Is because it is not a local email account. When you signed up with a Google account you never enter a password. Instead you're given a token that needs to be reauthentication. This will go through Google Services to reauthenticate the user and ultimately delete your account from FireBase.

**Comment by tonynguyen98** [comment]
> @dominikyz My task we to implement a service that will overwrite each user with "deleted". I believe the reason why we are doing this is to save the users ID and help identify which users needs to be deleted.

We're doing this so we can still have access to a deleted user's question and answers. So if a user deletes their account, we don't lose access the the `user_id` which is currently used in the question and answers table.

**Review by tonynguyen98 — APPROVED** [review]
Working on my end! Good work!

**Review by dominikyz — APPROVED** [review]
Thanks for explanation. Everything worked on my end just had couple questions. Great job!

**Pull request merged**